tests/test_update.py

```diff
@@ -663,12 +663,28 @@ def test_signed_updates(
                 shutil.rmtree(tmpdir, ignore_errors=True)
 
         put_no_sftp("image.mender", connection, remote="/data/image.mender")
+
+        # mender-convert'ed images don't have transient mender.conf
+        device_has_mender_conf = (
+            connection.run("test -f /etc/mender/mender.conf", warn=True).return_code
+            == 0
+        )
+        # mender-convert'ed images don't have this directory, but the test uses
+        # it to save certificates
+        connection.run("mkdir -p /data/etc/mender")
+
         try:
-            # Update key configuration on device.
-            connection.run(
-                "cp /etc/mender/mender.conf /data/etc/mender/mender.conf.bak"
-            )
-            get_no_sftp("/etc/mender/mender.conf", connection)
+            # Get configuration from device or create an empty one
+            if device_has_mender_conf:
+                connection.run(
+                    "cp /etc/mender/mender.conf /data/etc/mender/mender.conf.bak"
+                )
+                get_no_sftp("/etc/mender/mender.conf", connection)
+            else:
+                with open("mender.conf", "w") as fd:
+                    json.dump({}, fd)
```

Comment:
try/except here perhaps?

Reply:
Why? This is interned to create an empty file, so that the rest of the test can assume that the file exists and is json formatted.

Reply:
but if it fails, then the test will fail, right?

Reply:
Yes. But that is kind of the point, if it fails let's abort the test. What else can we do?